### PR TITLE
Menu: make link on the whole item

### PR DESCRIPTION
**Pull request by Krijovnick**

### CODE REVIEW

File: js/ui/context_menu/ui.menu_base.js [hunk 1]
```diff
@@ -177,18 +180,44 @@ class MenuBase extends HierarchicalCollectionWidget {
         this._initActions();
     }
 
+    _getLinkContainer(iconContainer, textContainer, { linkAttr, url }) {
+        if(iconContainer) {
+            iconContainer.addClass(DX_ICON_WITH_URL_CLASS);
```
Comment (ksercs) [inline js/ui/context_menu/ui.menu_base.js]: it's more compact to use optional chaining in such cases
```suggestion
            iconContainer?.addClass(DX_ICON_WITH_URL_CLASS);
```

File: js/ui/context_menu/ui.menu_base.js [hunk 2]
```diff
@@ -177,18 +180,44 @@ class MenuBase extends HierarchicalCollectionWidget {
...
+        }
+        $container.append(this._getPopoutContainer(itemData));
+        this._addContentClasses(itemData, $container.parent());
+    }
+
     _getTextContainer(itemData) {
-        const { text, url, linkAttr } = itemData;
+        const { text } = itemData;
         const $itemContainer = $('<span>').addClass(DX_MENU_ITEM_CAPTION_CLASS);
         const itemContent = isPlainObject(itemData) ? text : String(itemData);
```
Comment (ksercs) [inline js/ui/context_menu/ui.menu_base.js]: Now there can be a different type of content in item, so let's name it more specific
```suggestion
        const itemText = isPlainObject(itemData) ? text : String(itemData);
```

File: testing/tests/DevExpress.ui.widgets/menuBase.tests.js [hunk 3]
```diff
@@ -339,7 +343,7 @@ QUnit.module('Menu rendering', () => {
             const menuBase = createMenu({
                 items: [{ text: 'Item text', url }]
             });
-            const content = menuBase.element.find(`.${DX_MENU_ITEM_TEXT_CLASS}`).children()[0];
+            const content = menuBase.element.find(`.${DX_MENU_ITEM_CONTENT_CLASS}`).children()[0];
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets/menuBase.tests.js:346]: Why do we need this changes?

Comment (Krijovnick) [reply testing/tests/DevExpress.ui.widgets/menuBase.tests.js:346]: Because link in another place now

**Review by ksercs — APPROVED** [review]

**Review by marker-dao — APPROVED** [review]

**Pull request merged**